frontend/src/metabase/parameters/utils/parameter-values.js

```diff
@@ -10,11 +15,12 @@ export function getParameterValueFromQueryParams(parameter, queryParams) {
     return null;
   } else if (maybeParameterValue == null) {
     // try to use the default if the parameter is not present in the query params
-    return parameter.default ?? null;
-  } else {
-    const parsedValue = parseParameterValue(maybeParameterValue, parameter);
-    return normalizeParameterValueForWidget(parsedValue, parameter);
+    // then try to use recently used parameter
+    return parameter.default ?? recentlyUsedParameters[parameter.id] ?? null;
```

Comment:
I would suggest extracting this "default or last used" into a separate function and reusing it so it's one place where we decide about which value to use.